source/UIAHandler/remote.py

```diff
@@ -1,39 +1,150 @@
 # A part of NonVisual Desktop Access (NVDA)
 # This file is covered by the GNU General Public License.
 # See the file COPYING for more details.
-# Copyright (C) 2021-2022 NV Access Limited
+# Copyright (C) 2021-2024 NV Access Limited
 
 
-from typing import Optional, Union
-import os
-from ctypes import windll, byref, POINTER
-from comtypes.automation import VARIANT
-import NVDAHelper
+from typing import (
+	Optional,
+	Any,
+	Generator,
+	cast
+)
+from comtypes import GUID
 from comInterfaces import UIAutomationClient as UIA
+from ._remoteOps import remoteAlgorithms
+from ._remoteOps.remoteAPI import (
+	RemoteExtensionTarget,
+	RemoteInt
+)
+from ._remoteOps import operation
+from ._remoteOps import remoteAPI
+from ._remoteOps.lowLevel import (
+	TextUnit,
+	AttributeId,
+	StyleId
+)
 
 
-_dll = None
-
-
-def initialize(doRemote: bool, UIAClient: POINTER(UIA.IUIAutomation)):
+def initialize(doRemote: bool, UIAClient: UIA.IUIAutomation):
 	"""
 	Initializes UI Automation remote operations.
 	@param doRemote: true if code should be executed remotely, or false for locally.
 	@param UIAClient: the current instance of the UI Automation client library running in NVDA.
 	"""
-	global _dll
-	_dll = windll[os.path.join(NVDAHelper.versionedLibPath, "UIARemote.dll")]
-	_dll.initialize(doRemote, UIAClient)
+	return True
 
 
 def msWord_getCustomAttributeValue(
-		docElement: POINTER(UIA.IUIAutomationElement),
-		textRange: POINTER(UIA.IUIAutomationTextRange),
+		docElement: UIA.IUIAutomationElement,
+		textRange: UIA.IUIAutomationTextRange,
 		customAttribID: int
-) -> Optional[Union[int, str]]:
-	if _dll is None:
-		raise RuntimeError("UIARemote not initialized")
-	customAttribValue = VARIANT()
-	if _dll.msWord_getCustomAttributeValue(docElement, textRange, customAttribID, byref(customAttribValue)):
-		return customAttribValue.value
-	return None
+) -> Optional[Any]:

```

Comment:
is this an API breaking change (change of return type)

Reply:
The function will still return either an int or a string or None. In deed union of just int or string (not None) was originally incorrect.
Also, as execute is marked as always returning Any, then it is a bit tricky to really enforce  the right return type. I could cast the result of execute, but is it really worth it?

Reply:
I don't think that's needed, particularly if the original documentation was wrong